src/modules/flow/netctl/netctl.c

```diff
@@ -0,0 +1,327 @@
+connect_service_name(struct sol_flow_node *node, void *data, uint16_t port, uint16_t conn_id, const struct sol_flow_packet *packet)
+{
+    struct network_service_data *mdata = data;
+    struct sol_netctl_service *service;
+    int r;
+
+    service = sol_netctl_find_service_by_name(mdata->service_name);
+    if (!service) {
+        sol_flow_send_error_packet(node, ENOENT, "Did not found service name %s", mdata->service_name);
+        return 0;
+    }
+
+    r = sol_netctl_service_connect(service);
+    if (r < 0)
+        sol_flow_send_error_packet(node, -r, "Could not connect to service: %s: %s",
+            mdata->service_name, sol_util_strerrora(-r));
+
+    return 0;
+}
+
+static int
+disconnect_service_name(struct sol_flow_node *node, void *data, uint16_t port, uint16_t conn_id, const struct sol_flow_packet *packet)
+{
+    struct network_service_data *mdata = data;
+    struct sol_netctl_service *service;
+    int r;
+
+    service = sol_netctl_find_service_by_name(mdata->service_name);
+    if (!service) {
+        sol_flow_send_error_packet(node, ENOENT, "Did not found service name %s", mdata->service_name);
+        return 0;
+    }
+
+    r = sol_netctl_service_disconnect(service);
+    if (r < 0)
+        sol_flow_send_error_packet(node, -r, "Could not disconnect to service: %s: %s",
+            mdata->service_name, sol_util_strerrora(-r));
+
+    return 0;
+}
+
+static int
+state_dispatch(struct sol_flow_node *node, void *data)
+{
+    struct network_service_data *mdata = data;
+    struct sol_netctl_service *service;
+    int r;
+
+    service = sol_netctl_find_service_by_name(mdata->service_name);
+    if (!service) {
+        sol_flow_send_error_packet(node, ENOENT, "Did not found service name %s", mdata->service_name);
+        return 0;
+    }
+    r = sol_flow_send_string_packet(node,
+        SOL_FLOW_NODE_TYPE_NETCTL_SERVICE__OUT__OUT,
+        sol_netctl_service_state_to_str(mdata->state));
+    SOL_INT_CHECK(r, < 0, r);
+
+    if (mdata->state == SOL_NETCTL_SERVICE_STATE_ONLINE)
+        r = sol_flow_send_bool_packet(node,
+            SOL_FLOW_NODE_TYPE_NETCTL_SERVICE__OUT__ONLINE,
+            true);
+    else
+        r = sol_flow_send_bool_packet(node,
+            SOL_FLOW_NODE_TYPE_NETCTL_SERVICE__OUT__ONLINE,
+            false);
+
+    SOL_INT_CHECK(r, < 0, r);
+
+    return 0;
+}
+
+static int
+get_service_state(struct sol_flow_node *node, void *data, uint16_t port, uint16_t conn_id, const struct sol_flow_packet *packet)
+{
+    return state_dispatch(node, data);
+}
+
+static int
+set_service_name(struct sol_flow_node *node, void *data, uint16_t port, uint16_t conn_id, const struct sol_flow_packet *packet)
+{
+    struct network_service_data *mdata = data;
+    const char *in_value;
+    int r;
+
+    r = sol_flow_packet_get_string(packet, &in_value);
+    SOL_INT_CHECK(r, < 0, r);
+
+    sol_util_replace_str_if_changed(&mdata->service_name, in_value);
+    SOL_NULL_CHECK(mdata->service_name, -ENOMEM);
```

Comment:
`sol_util_replace_str_if_changed()` already returns error, you can use that. Also note that if `return == 0`, then it did not change, so no need to process the rest of this function.